ibis/backends/dask/core.py

```diff
@@ -0,0 +1,547 @@
+"""The dask backend is a very close port of the pandas backend, and thus
+has the same caveats.
+
+The dask backend is a departure from the typical ibis backend in that it
+doesn't compile to anything, and the execution of the ibis expression
+is under the purview of ibis itself rather than executing SQL on a server.
+
+Design
+------
+The dask backend uses a technique called `multiple dispatch
+<https://en.wikipedia.org/wiki/Multiple_dispatch>`_, implemented in a
+third-party open source library called `multipledispatch
+<https://github.com/mrocklin/multipledispatch>`_.
+
+Multiple dispatch is a generalization of standard single-dispatch runtime
+polymorphism to multiple arguments.
+
+Compilation
+-----------
+This is a no-op because we execute ibis expressions directly.
```

Comment:
hmm this is pretty misleading, dask *does* compile to a dask expression, and then *execute* computes on it, unless pandas which doesn't have a compliation step at all. this is much more akin to the way spark works, no?

Reply:
I can take a look at what pyspark does. 

I think the point I was trying to make here is that ibis-dask will ultimately pass you back a dask expression that you can either `.compute()` (giving you a `pd.DataFrame`) or continue to run computations on. So we're using ibis to build up the expression and it is ultimately up the user to compute that expression, rather than doing it for them.

Reply:
yes that's exactly right, i think we just want to explain that in the documentation (e.g. here) 'execute ibis expressions directly' is true, but a bit misleading (e.g. say what you wrote right above)